contracts/Ally.sol

```diff
@@ -1,113 +0,0 @@
-// SPDX-License-Identifier: BUSL-1.1
-
-pragma solidity 0.7.6;
-
-import "@openzeppelin/contracts/token/ERC20/ERC20Burnable.sol";
-import "@openzeppelin/contracts/token/ERC20/IERC20.sol";
-import "@openzeppelin/contracts/token/ERC20/SafeERC20.sol";
-import "@openzeppelin/contracts/math/SafeMath.sol";
-import "@openzeppelin/contracts/access/Ownable.sol";
-
-contract Ally is ERC20Burnable, Ownable {
-
-    using SafeMath for uint256;
-    using SafeERC20 for IERC20;
-
-    IERC20 ichi;
-
-    string private constant NAME_ = "ICHI Ally";
-    string private constant SYMBOL_ = "ALLY";
-    uint256 private constant INITIAL_SUPPLY_ = 10000000 * 1e18;
-    uint256 private constant PRECISION = 1e18;
-
-    // start time and duration are unchangeable
-
-    uint256 public immutable commencement;
-    uint256 public immutable durationSeconds;
-
-    event Deployed(address deployer, IERC20 ichi, uint256 commencement, uint256 durationDays);
-    event ClaimIchi(address user, address to, uint256 ally, uint256 ichi);
-    event EmergencyWithdrawal(IERC20 token, uint256 amount, address to);
-
-    /// @notice following the airdrop, the deployer is required to burn any surplus Ally
-    /// (because totalSupply() is used in calculations) and send ICHI to the contract.
-
-    constructor(IERC20 ichi_, uint256 commencement_, uint256 durationDays_)
-        ERC20(NAME_, SYMBOL_) 
-    {
-        require(commencement_ >= block.timestamp, 'Ally:constructor:: commencement_ cannot be in the past');
-        ichi = ichi_;
-        commencement = commencement_;
-        durationSeconds = durationDays_.mul(1 days);
-        _mint(msg.sender, INITIAL_SUPPLY_);
-        emit Deployed(msg.sender, ichi_, commencement_, durationDays_);
-    }
-
-    // redeem ichi from this contract by burning Ally
-
-    function claimIchi(uint256 amountAlly, address to) external returns(uint256 amountIchi) {
-        require(commencement < block.timestamp, 'Ally:claimIchi:: must wait for the commencement date to pass');
-        require(to != address(0), "Ally:claimIchi:: to cannot be the 0x0 address");
-        require(amountAlly <= balanceOf(msg.sender), 'Ally:claimIchi:: insufficient Ally balance');
-        require(amountAlly <= allowance(msg.sender, address(this)), 'Ally:claimIchi:: insufficent Ally allowance');
```

Comment:
@vastreck also I removed this line - it doesn't really make sense for the user to have to approve this contract to transfer the token, since this contract is the token

Reply:
this is in reference to the line:
`require(amountAlly <= allowance(msg.sender, address(this)), 'Ally:claimIchi:: insufficent Ally allowance');`

Reply:
agree